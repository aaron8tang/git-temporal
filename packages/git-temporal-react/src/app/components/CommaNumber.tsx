import React from 'react';
import { style } from 'app/styles';

export interface NumberProps {
  // This is the text or JSX that gets wrapped in stacked label
  value: string | number;
  style?: object | string;
}

export function convertNumberToStringWithCommas(numberValue: string | number) {
  let numericValue = parseFloat(numberValue as string);
  if (isNaN(numericValue)) {
    numericValue = 0;
  }
  const [wholeNumber, decimal] = numericValue.toString().split('.');
  let convertedValue = wholeNumber.replace(/\B(?=(\d{3})+(?!\d))/g, ',');
  if (decimal) {
    convertedValue += `.${decimal}`;
  }
  return convertedValue;
}
// numbers for humans have commas
export const CommaNumber = (props: NumberProps): JSX.Element => {
  return (
    <span style={style(props.style)}>
      {convertNumberToStringWithCommas(props.value)}
    </span>
  );
};

CommaNumber.displayName = 'CommaNumber';
