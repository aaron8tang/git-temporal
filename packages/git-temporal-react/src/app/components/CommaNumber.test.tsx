import React from 'react';
import { shallow } from 'enzyme';

import { CommaNumber } from './CommaNumber';

describe('components/CommaNumber', () => {
  describe('when rendered with value as string', () => {
    let wrapper;
    beforeAll(() => {
      wrapper = shallow(<CommaNumber value="100545007.02" />);
    });

    test('it should match snapshot (it should have rendered 100,545,007.02)', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('when rendered with value as numeric', () => {
    let wrapper;
    beforeAll(() => {
      wrapper = shallow(<CommaNumber value={100545007.02} />);
    });

    test('it should match snapshot (it should have rendered 100,545,007.02)', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('when rendered with non numeric value as string', () => {
    let wrapper;
    beforeAll(() => {
      wrapper = shallow(<CommaNumber value="NaN is not a number" />);
    });

    test('it should match snapshot (it should have rendered 0)', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
