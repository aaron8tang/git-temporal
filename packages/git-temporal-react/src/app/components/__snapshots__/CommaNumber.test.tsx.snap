// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/CommaNumber when rendered with non numeric value as string it should match snapshot (it should have rendered 0) 1`] = `
<span
  style={Object {}}
>
  0
</span>
`;

exports[`components/CommaNumber when rendered with value as numeric it should match snapshot (it should have rendered 100,545,007.02) 1`] = `
<span
  style={Object {}}
>
  100,545,007.02
</span>
`;

exports[`components/CommaNumber when rendered with value as string it should match snapshot (it should have rendered 100,545,007.02) 1`] = `
<span
  style={Object {}}
>
  100,545,007.02
</span>
`;
